arm/Microsoft.Synapse/workspaces/keys/deploy.bicep

```diff
@@ -31,6 +58,31 @@ resource key 'Microsoft.Synapse/workspaces/keys@2021-06-01' = {
   }
 }
 
+resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (enableDefaultTelemetry) {
+  name: 'pid-47ed15a6-730a-4827-bcb4-0fd963ffbd82-${uniqueString(deployment().name, location)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
+
+// Diagnostics Settings
+resource workspace_diagnosticSettings 'Microsoft.Insights/diagnosticsettings@2021-05-01-preview' = if (!empty(diagnosticStorageAccountId) || !empty(diagnosticWorkspaceId) || !empty(diagnosticEventHubAuthorizationRuleId) || !empty(diagnosticEventHubName)) {
```

Comment:
I don't think this makes sense. The diagnostic settings for the workspace are set in the main `deploy.bicep` and should not be set in a child template - unless it itself supports diagnostic settings (e.g. `key_diagnosticSettings` with `scope:`key`).

Reply:
I'd hence suggest to remove the diagnostic settings form this template and just keep it in the parent